src/main/java/seedu/address/logic/parser/finance/Payparser.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.logic.parser.finance;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GENDER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_JOIN_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PAY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.finance.Pay;
+import seedu.address.logic.commands.finance.Pay.EditEmployeeDescriptor;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new Pay object
+ */
+public class Payparser implements Parser<Pay> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the Pay
+     * and returns an Pay object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public Pay parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
```

Comment:
There isn't a need for all this, which is just a repeat of the functionality of the `EditEmployeeCommand` code. This violates the Single Responsibility Principle - if I needed to edit pay I should only need to call `EditEmployeeCommand`.